### PR TITLE
refactor(connector): [Dlocal] Currency Unit Conversion

### PR DESCRIPTION
**Pull request by SagarDevAchar**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Addressing Issue: #2223 
- Modified two files in `hyperswitch/crates/router/src/connector/`
  - `dlocal.rs`
    - Implement `get_currency_unit` function
    - Modify `ConnectorIntegration` implementations for `Dlocal`
  - `dlocal/transformers.rs`
    - Implement `DlocalRouterData<T>` structure and functionality
    - Modify implementation for `DlocalPaymentsRequest`

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by Sangamesh26 — DISMISSED** [review]
LGTM :)

File: crates/router/src/connector/dlocal/transformers.rs [hunk 1]
```diff
@@ -399,22 +439,24 @@ impl<F, T>
...
 
-impl<F> TryFrom<&types::RefundsRouterData<F>> for RefundRequest {
+impl<F> TryFrom<&DlocalRouterData<&types::RefundsRouterData<F>>> for DlocalRefundRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
-        let amount_to_refund = item.request.refund_amount.to_string();
+    fn try_from(
+        item: &DlocalRouterData<&types::RefundsRouterData<F>>,
+    ) -> Result<Self, Self::Error> {
+        let amount_to_refund = item.router_data.request.refund_amount.to_string();
```
Comment (srujanchikke) [inline crates/router/src/connector/dlocal/transformers.rs:454]:
```suggestion
        let amount_to_refund = item.amount.clone();
```

File: crates/router/src/connector/dlocal/transformers.rs [hunk 2]
```diff
@@ -66,22 +98,24 @@ pub struct DlocalPaymentsRequest {
...
             api::PaymentMethodData::Card(ref ccard) => {
                 let should_capture = matches!(
-                    item.request.capture_method,
+                    item.router_data.request.capture_method,
                     Some(enums::CaptureMethod::Automatic)
                 );
                 let payment_request = Self {
-                    amount: item.request.amount,
-                    currency: item.request.currency,
+                    amount: item.router_data.request.amount,
```
Comment (srujanchikke) [inline crates/router/src/connector/dlocal/transformers.rs:117]:
```suggestion
                    amount: item.amount.clone(),
```

File: crates/router/src/connector/dlocal/transformers.rs (hunk 1 above)
Comment (SagarDevAchar) [reply crates/router/src/connector/dlocal/transformers.rs:454]: Updated, thanks!

**Comment by SagarDevAchar** [comment]
@srujanchikke @swangi-kumari 
I need some help with the `amount` field value. I wanted to make sure I get it right before I make another commit.

This is what I presume the line should be based on my understanding of `crates\router\src\connector\utils.rs`:
```
117 |    amount: utils::get_amount_as_string(CURRENCY_UNIT, item.amount.clone(), CURRENCY)?,
```
- What is the field I need to be passing for `CURRENCY_UNIT`? 
  - Should I just pass `types::api::CurrencyUnit::Base`?
- What is the field I need to be passing for `CURRENCY`?
  - I assume it is `item.router_data.request.currency`

**Comment by SagarDevAchar** [comment]
@srujanchikke 
My bad, the `amount` field is expecting an `i64` while `item.amount` is a `String`.
But in `crates\router\src\connector\utils.rs` I couldn't find a suitable function to achieve this.

What should I do now?

**Comment by srujanchikke** [comment]
> @srujanchikke My bad, the `amount` field is expecting an `i64` while `item.amount` is a `String`. But in `utils.rs` I couldn't find a suitable function to achieve this. Should I just use `str::parse()`?
> 
> Also, can you confirm if I have correctly defined the `get_currency_unit()` in `dlocal.rs` with `api::CurrencyUnit::Base`? Seeing the `i64` requirement, I suspected that I had to use `api::CurrencyUnit::Minor`

if amount is i64 then you may need to make amount in `DlocalRoutrerData` as i64 . You can refer to [checkout](https://github.com/juspay/hyperswitch/pull/2192/files) files

**Review by srujanchikke — APPROVED** [review]
LGTM

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @SagarDevAchar!

**Comment by swangi-kumari** [comment]
Hey @SagarDevAchar ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**